etl/steps/data/garden/wb/2025-04-08/world_bank_pip_legacy.py

```diff
@@ -0,0 +1,1465 @@
+
+    # Keep columns needed
+    tb_survey = tb_survey[["country", "year", "surveys_past_decade"]]
+
+    # Merge with original table
+    tb = pr.merge(tb_survey, tb, on=["country", "year"], how="outer")
+
+    return tb
+
+
+def drop_columns(tb: Table) -> Table:
+    """
+    Drop columns not needed
+    """
+
+    # Remove columns
+    tb = tb.drop(
+        columns=[
+            "ppp_version",
+            "reporting_pop",
+            "is_interpolated",
+            "distribution_type",
+            "estimation_type",
+            "survey_comparability",
+            "comparable_spell",
+        ]
+    )
+
+    return tb
+
+
+def create_survey_spells(tb: Table) -> list:
+    """
+    Create tables for each indicator and survey spells, to be able to graph them in explorers.
+    """
+
+    tb = tb.copy()
+
+    # drop rows where survey coverage = nan (This is just regions)
+    tb = tb[tb["survey_comparability"].notna()].reset_index()
+
+    # Add 1 to make comparability var run from 1, not from 0
+    tb["survey_comparability"] += 1
+
+    # Note the welfare type in the comparability spell
+    tb["survey_comparability"] = (
+        tb["welfare_type"].astype(str) + "_spell_" + tb["survey_comparability"].astype(int).astype(str)
+    )
+
+    # Remove columns not needed: stacked, above, etc
+    drop_list = ["above", "between", "poverty_severity", "watts"]
+    for var in drop_list:
+        tb = tb[tb.columns.drop(list(tb.filter(like=var)))]
+
+    vars = [
+        i
+        for i in tb.columns
+        if i
+        not in [
+            "country",
+            "year",
+            "ppp_version",
+            "reporting_level",
+            "welfare_type",
+            "reporting_pop",
+            "is_interpolated",
+            "distribution_type",
+            "estimation_type",
+            "survey_comparability",
+            "comparable_spell",
+            f"headcount_{INTERNATIONAL_POVERTY_LINE_CURRENT}_regions",
+            "surveys_past_decade",
+        ]
+    ]
+
+    # Define spell table list
+    spell_tables = []
+
+    # Loop over the variables in the main dataset
+    for select_var in vars:
+        tb_var = tb[["country", "year", select_var, "survey_comparability"]].copy()
+
+        # convert to wide
+        tb_var = pr.pivot(
+            tb_var,
+            index=["country", "year"],
+            columns=["survey_comparability"],
+            values=select_var,
+        )
+
+        tb_var.metadata.short_name = f"{tb_var.metadata.short_name}_{select_var}"
+
+        spell_tables.append(tb_var)
+
+    return spell_tables
+
+
+def create_survey_spells_inc_cons(tb_inc: Table, tb_cons: Table) -> list:
+    """
+    Create table for each indicator and survey spells, to be able to graph them in explorers.
+    This version recombines income and consumption tables to not lose dropped rows.
+    """
+
+    tb_inc = tb_inc.reset_index().copy()
+    tb_cons = tb_cons.reset_index().copy()
```

Comment:
I've seen various instances of `reset_index().copy()`, and I think the `copy()` can be dropped, since `reset_index()` creates a copy already.